From 8f8fc836709cff58bd666b87937542088cfc19db Mon Sep 17 00:00:00 2001
From: Kalev Lember <kalevlember@gmail.com>
Date: Wed, 28 Mar 2012 12:42:49 +0300
Subject: [PATCH] configure: Fix alignment tests when cross compiling

When inserting custom code to AC_CHECK_ALIGNOF, make sure to not replace
the default includes, but instead append to them.

This fixes ALIGNOF_GUINT32 and ALIGNOF_GUINT64 that were both 0 when cross
compiling. The third 'unsigned long' test wasn't affected because the
AC_CHECK_ALIGNOF call didn't specify the optional 2nd parameter.
---
 configure.ac |    6 ++++--
 1 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 0d30b3e..82d74c6 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3405,8 +3405,10 @@ $ac_cv_sizeof___int64)
   ;;
 esac
 
-AC_CHECK_ALIGNOF([guint32], [typedef unsigned $gint32 guint32;])
-AC_CHECK_ALIGNOF([guint64], typedef unsigned $gint64 guint64;)
+AC_CHECK_ALIGNOF([guint32], [AC_INCLUDES_DEFAULT
+typedef unsigned $gint32 guint32;])
+AC_CHECK_ALIGNOF([guint64], [AC_INCLUDES_DEFAULT
+typedef unsigned $gint64 guint64;])
 AC_CHECK_ALIGNOF([unsigned long])
 
 # Check for libdbus1 - Optional - is only used in the GDBus test cases
-- 
1.7.9.1

